### PR TITLE
#3 - Add Database connection system and  #4 - Save the HVAC data in the database

### LINKED ISSUE #3: Add Database connection system
Add a way to create a database connection (the same DB than Lab1). 

The project must connect to the db only once, during the first execution and **not** before each metric sending.

From the Lab2 presentation :
```
Vous devrez initialiser la connexion à la base de données lors du démarrage de l’application et
non à chaque envoi de données (afin d’éviter de créer un grand nombre de connexions). L’endroit
précis est indiqué en commentaire dans le code pour vous aider.
```

### LINKED ISSUE #4: Save the HVAC data in the database
Blocked by #3 

Add the feature to send the HVAC data to the connected database.

From the Lab2 presentation :
```
Vous devrez modifier l’application HVAC afin de pouvoir sauvegarder les informations suivantes
dans la base de données du laboratoire (une table différente de celle du laboratoire 1) :
- Les températures reçues et l’heure à laquelle elles ont été reçues
- Les événements liés à votre application HVAC tels que les activations de la climatisation
ou du chauffage
```

### PR DESCRIPTION
**Pull request by lanahria**
# Description

Adding connection to the databse, sending Hvac data to it as well as logs from actions into the database.

Closes #3
Closes #4 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import json
 import logging
 import time
-from signalrcore.hub_connection_builder import HubConnectionBuilder
+from urllib.parse import urlparse
```
Comment (RenaldMorino) [inline src/main.py:4]: Faudrait tu rajouter ça dans les dependencies du pipfile?

File: src/main.py [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 import requests
+import psycopg2
+from psycopg2 import OperationalError, DatabaseError
+from signalrcore.hub_connection_builder import HubConnectionBuilder
```
Comment (RenaldMorino) [inline src/main.py:8]: Pis ça?

**Review by RenaldMorino — DISMISSED** [review]

File: src/main.py (hunk 1 above)
Comment (lanahria) [reply src/main.py:4]: Oui ! J'avais oublié

**Review by RenaldMorino — DISMISSED** [review]
Looks good

File: .gitignore [hunk 3]
```diff
@@ -136,4 +136,5 @@ deploy_rsa
 deploy_rsa.pub
 
 # Kubernetes config
-kubeconfig.yaml
\ No newline at end of file
+kubeconfig.yaml
+Pipfile.lock
```
Comment (Arachnight) [inline .gitignore]: Faut pas ignorer le Pipfile.lock.
C'est ce qui permet d'installer en prod les librairies nécessaires, en bloquant les versions des libraires, pour ne pas avoir un décalage de version

**Review by Arachnight — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (lanahria) [reply .gitignore]: Mais le pipeline fait pas un pipenv install ?
Si oui, ca sert a rien que ce fichier, generer et modifier par chaque utilisateur, soit sur le depot.

**Review by Arachnight — APPROVED** [review]

**Pull request merged**